aws/ec2_utils.go

```diff
@@ -0,0 +1,155 @@
+package aws
+
+import (
+	"fmt"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	gruntworkerrors "github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+func getAMIIdByName(svc *ec2.EC2, name string) (string, error) {
+	imagesResult, err := svc.DescribeImages(&ec2.DescribeImagesInput{
+		Owners: []*string{awsgo.String("self"), awsgo.String("amazon")},
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("name"),
+				Values: []*string{awsgo.String(name)},
+			},
+		},
+	})
+
+	if err != nil {
+		return "", gruntworkerrors.WithStackTrace(err)
+	}
+
+	if len(imagesResult.Images) == 0 {
+		return "", gruntworkerrors.WithStackTrace(fmt.Errorf("No images found with name %s", name))
+	}
+
+	image := imagesResult.Images[0]
+	return awsgo.StringValue(image.ImageId), nil
+}
+
+// runAndWaitForInstance - Given a preconstructed ec2.RunInstancesInput object,
+// make the API call to run the instance and then wait for the instance to be
+// up and running before returning.
+func runAndWaitForInstance(svc *ec2.EC2, name string, params *ec2.RunInstancesInput) (ec2.Instance, error) {
+	runResult, err := svc.RunInstances(params)
+	if err != nil {
+		return ec2.Instance{}, gruntworkerrors.WithStackTrace(err)
+	}
+
+	if len(runResult.Instances) == 0 {
+		err := fmt.Errorf("Could not create test EC2 instance")
+		return ec2.Instance{}, gruntworkerrors.WithStackTrace(err)
+	}
+
+	err = svc.WaitUntilInstanceExists(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("instance-id"),
+				Values: []*string{runResult.Instances[0].InstanceId},
+			},
+		},
+	})
+
+	if err != nil {
+		return ec2.Instance{}, gruntworkerrors.WithStackTrace(err)
+	}
+
+	// Add test tag to the created instance
+	_, err = svc.CreateTags(&ec2.CreateTagsInput{
+		Resources: []*string{runResult.Instances[0].InstanceId},
+		Tags: []*ec2.Tag{
+			{
+				Key:   awsgo.String("Name"),
+				Value: awsgo.String(name),
+			},
+		},
+	})
+
+	if err != nil {
+		return ec2.Instance{}, gruntworkerrors.WithStackTrace(err)
+	}
+
+	// EC2 Instance must be in a running before this function returns
+	err = svc.WaitUntilInstanceRunning(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("instance-id"),
+				Values: []*string{runResult.Instances[0].InstanceId},
+			},
+		},
+	})
+
+	if err != nil {
+		return ec2.Instance{}, gruntworkerrors.WithStackTrace(err)
+	}
+
+	return *runResult.Instances[0], nil
+
+}
+
+// CreateTestEC2Instance - creates a test EC2 instance
+func CreateTestEC2Instance(session *session.Session, name string, protected bool) (ec2.Instance, error) {
```

Comment:
This is no longer in `ec2_test.go` as it is moved to `ec2_utils.go` because `cli_nukeallresources_test.go` creates test S3 and test EC2 instance and we needed that logic to be moved out of test files as per. earlier discssion. So I moved the corresponding utility functions in `ec2_test.go` and `s3_test.go`. Going by the normal convention of creating an AWS resource or returning error if there is a failure. Cannot do an `assert` or a `require` in this method. Does that make sense or is there a better way around this?